src/gui/painting/qcosmeticstroker.cpp

```diff
@@ -97,6 +100,61 @@ struct Dasher {
     }
 };
 
+struct FixedDasher
+{
+    QCosmeticStroker *stroker;
+    int *pattern;
+    qlonglong offset;
+    int dashIndex;
+    int dashOn;
+    int seenOn = 0;
+    int dashInc;
+
+    FixedDasher(QCosmeticStroker *s, bool reverse, int start, int stop, int dash_offset,
+                int dash_inc)
+        : stroker(s), dashInc(dash_inc)
+    {
+        int delta = stop - start;
+        if (reverse) {
+            pattern = stroker->reversePattern;
+            offset = qlonglong(stroker->patternOffset - dash_offset) << 10;
+            dashOn = 0;
+        } else {
+            pattern = stroker->pattern;
+            offset = qlonglong(stroker->patternOffset + dash_offset) << 10;
+            dashOn = 1;
+        }
+        offset %= qlonglong(stroker->patternLength) << 10;
+        if (offset < 0)
+            offset += qlonglong(stroker->patternLength) << 10;
+
+        dashIndex = 0;
+        while (offset >= qlonglong(pattern[dashIndex]) << 10)
+            ++dashIndex;
+
+        //  qDebug() << "   dasher" << offset / 64. << reverse << dashIndex << offset << stroker->patternLength;
+        stroker->patternOffset += delta;
+        stroker->patternOffset %= stroker->patternLength;
+    }
+
+    bool on() const { return seenOn | ((dashIndex + dashOn) & 1); }
```

Comment:
Yeah, I thought it might be a tiny bit quicker to use the already-known value first and do short-circuit evaluation, avoiding doing the addition and `& 1`. Although looking at it now, it's currently a binary-or rather than a logical-or, so it won't actually do that anyway

Reply:
yeah, it should not make a difference performance wise. Okay, I thought I'm missing something :)